docs/developer-guide/implement-a-udf.rst

```diff
@@ -0,0 +1,349 @@
+        <version>1.0</version>
+
+        <!-- Specify the repository for Confluent dependencies -->
+        <repositories>
+            <repository>
+                <id>confluent</id>
+                <url>http://packages.confluent.io/maven/</url>
+            </repository>
+        </repositories>
+
+        <!-- Specify build properties -->
+        <properties>
+            <exec.mainClass>io.confluent.ksql.udfdemo.thisisignored</exec.mainClass>
+            <java.version>1.8</java.version>
+            <kafka.version>2.0.0</kafka.version>
+            <kafka.scala.version>2.11</kafka.scala.version>
+            <scala.version>${kafka.scala.version}.8</scala.version>
+            <confluent.version>5.1.0</confluent.version>
+            <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        </properties>
+
+        <!-- Specify the ksql-udf dependency -->
+        <dependencies>
+            <!-- KSQL dependency is needed to write your own UDF -->
+            <dependency>
+                <groupId>io.confluent.ksql</groupId>
+                <artifactId>ksql-udf</artifactId>
+                <version>|release|</version>
+            </dependency>
+        </dependencies>
+
+        <!-- Build boilerplate -->
+        <build>
+            <plugins>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-compiler-plugin</artifactId>
+                    <version>3.6.1</version>
+                    <configuration>
+                        <source>1.8</source>
+                        <target>1.8</target>
+                    </configuration>
+                </plugin>
+
+                <!-- Package all dependencies as one jar -->
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-assembly-plugin</artifactId>
+                    <version>2.5.2</version>
+                    <configuration>
+                        <descriptorRefs>
+                            <descriptorRef>jar-with-dependencies</descriptorRef>
+                        </descriptorRefs>
+                        <archive>
+                            <manifest>
+                                <addClasspath>true</addClasspath>
+                                <mainClass>${exec.mainClass}</mainClass>
+                            </manifest>
+                        </archive>
+                    </configuration>
+                    <executions>
+                        <execution>
+                            <id>assemble-all</id>
+                            <phase>package</phase>
+                            <goals>
+                                <goal>single</goal>
+                            </goals>
+                        </execution>
+                    </executions>
+                </plugin>
+            </plugins>
+        </build>
+    </project>
+
+Build the UDF Package

```

Comment:
Is it worth having a step about testing the code, or at least a comment saying we _highly_ recommend the write comprehensive tests to cover their UDFs.